### PR TITLE
Add ServiceAccount doc

### LINKED ISSUE #2521: [opentelemetry-operator] - Add support for imagePullSecrets for collectors
I encounter a scenario where I must retrieve images from a private registry. While I successfully utilized the "imagePullSecrets" feature within the helm chart for the otel-operator, I encountered difficulty in figuring out how to apply the same mechanism to the collectors generated by the otel-operator.

### PR DESCRIPTION
**Pull request by yuriolisa**
**Description:** <Describe what has changed.>

Added documentation for use ServiceAccount imagePullSecrets
**Link to tracking Issue:** <Issue number if applicable>
Resolves #2521 
**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: .chloggen/add-imagepullsecrets-doc.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'enhancement'
```
Comment (pavolloffay) [inline .chloggen/add-imagepullsecrets-doc.yaml:2]: Can we remove the entry. We are not adding any new feature in this patch

File: README.md [hunk 2]
```diff
@@ -665,6 +665,55 @@ spec:
           exporters: [debug]
 ```
 
+#### Using imagePullSecrets
```
Comment (pavolloffay) [inline README.md:668]: This is being added under TA. It should rather belong to the general collector section.

File: README.md [hunk 3]
```diff
@@ -665,6 +665,55 @@ spec:
...
+```yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: private-collector
+spec:
+  mode: deployment
+  image: 
+  serviceAccount: <service-account-name>
+  config: |
```
Comment (pavolloffay) [inline README.md:701]: I would skip this section as it is irrelevant in this context

File: README.md [hunk 4]
```diff
@@ -665,6 +665,55 @@ spec:
...
+kubectl create serviceaccount <service-account-name>
+````
+
+* Create an imagePullSecret.
+
+````bash
+kubectl create secret docker-registry <secret-name> --docker-server=<registry name> \
+        --docker-username=DUMMY_USERNAME --docker-password=DUMMY_DOCKER_PASSWORD \
+        --docker-email=DUMMY_DOCKER_EMAIL
+
```
Comment (pavolloffay) [inline README.md:683]: remove empty line

Comment (yuriolisa) [reply README.md]: @pavolloffay , I made all the requested changes.

**Review by jaronoff97 — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**